### PR TITLE
feat(keycloak): add keycloak module

### PR DESCRIPTION
**Pull request by xCryzed**
## Information

| Type                  | Breaking change |
| --------------------- | --------------- |
| Feature | No          |

---

WIP

### CODE REVIEW

File: packages/third-parties/keycloak/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "@tsed/keycloak",
+  "version": "6.102.7",
+  "description": "Keycloak package for Ts.ED framework",
+  "source": "./src/index.ts",
+  "main": "./lib/cjs/index.js",
```
Comment (Romakita) [inline packages/third-parties/keycloak/package.json:6]: "module": "./lib/esm/index.js"  

I'll add it for other modules in another PR ;)

File: packages/third-parties/keycloak/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  "private": false,
+  "devDependencies": {
+    "@tsed/common": "6.102.7",
+    "@tsed/core": "6.102.7",
+    "@tsed/di": "6.102.7",
+    "keycloak-connect": "17.0.0"
+  },
+  "peerDependencies": {
+    "keycloak-connect": "17.0.0"
```
Comment (Romakita) [inline packages/third-parties/keycloak/package.json]: "keycloak-connect": ">=17.0.0"

File: packages/third-parties/keycloak/src/KeycloakModule.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  async $afterListen(): Promise<any> {
+    if (this.loadKeycloak) {
+      this.logger.info("Initialize Keycloak adapter...");
+      this.keycloak = new KeycloakConnect(this.kcOptions, this.kcConfig);
+    }
+  }
+
+  $onDestroy(): Promise<any> | void {
+    return undefined;
```
Comment (Romakita) [inline packages/third-parties/keycloak/src/KeycloakModule.ts]: probably, it's not finish here ^^

File: packages/third-parties/keycloak/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    interface Configuration {
+      keycloak?: {
+        /**
+         * Enable Keycloak. Default false.
+         */
+        enabled?: boolean;
+        /**
+         * Keycloak options for web session store, scopes and cookies. Default empty object.
+         */
+        kcOptions?: KeycloakOptions;
```
Comment (Romakita) [inline packages/third-parties/keycloak/src/index.ts]: options?: KeycloakOptions;

where are already in keycloak objet ^^

File: packages/third-parties/keycloak/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        /**
+         * Keycloak configuration or the path to the configuration file.
+         */
+        kcConfig: KeycloakConfig | string;
```
Comment (Romakita) [inline packages/third-parties/keycloak/src/index.ts]: config: KeycloakConfig | string;

**Review by Romakita — CHANGES_REQUESTED** [review]

File: packages/third-parties/keycloak/package.json (hunk 2 above)
Comment (xCryzed) [reply packages/third-parties/keycloak/package.json]: Done

File: packages/third-parties/keycloak/README.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+   <h1>Keycloak</h1>
+
+[![Build & Release](https://github.com/tsedio/tsed/workflows/Build%20&%20Release/badge.svg)](https://github.com/tsedio/tsed/actions?query=workflow%3A%22Build+%26+Release%22)
+[![PR Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/tsedio/tsed/blob/master/CONTRIBUTING.md)
+[![Coverage Status](https://coveralls.io/repos/github/tsedio/tsed/badge.svg?branch=production)](https://coveralls.io/github/tsedio/tsed?branch=production)
+[![npm version](https://badge.fury.io/js/%40tsed%2Fcommon.svg)](https://badge.fury.io/js/%40tsed%2Fcommon)
+[![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
+[![code style: prettier](https://img.shields.io/badge/code_style-prettier-ff69b4.svg?style=flat-square)](https://github.com/prettier/prettier)
+[![backers](https://opencollective.com/tsed/tiers/badge.svg)](https://opencollective.com/tsed)
+
```
Comment (Romakita) [inline packages/third-parties/keycloak/README.md:16]: The readme is incomplete :)

Comment (Romakita) [reply packages/third-parties/keycloak/README.md:16]: You can take the stripe readme as example ^^

File: packages/third-parties/keycloak/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import {KeycloakConfig, KeycloakOptions} from "keycloak-connect";
```
Comment (Romakita) [inline packages/third-parties/keycloak/src/index.ts]: can you move the `declare global` code part in `interfaces/interfaces.ts`. The RC branch change the directory structures do be compliant with ESM module :)

**Review by Romakita — CHANGES_REQUESTED** [review]

**Review by Romakita — APPROVED** [review]

**Comment by Romakita** [comment]
@xCryzed is it ready?

**Comment by xCryzed** [comment]
@Romakita we still need tests and a little bit more documenation.

**Pull request closed without merging**